src/lib/Specification/VersionHasConflict.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Specification;
+
+use eZ\Publish\API\Repository\Exceptions\UnauthorizedException;
+use eZ\Publish\Core\Base\Exceptions\BadStateException;
+use eZ\Publish\API\Repository\ContentService;
+use eZ\Publish\API\Repository\Values\Content\VersionInfo;
+
+class VersionHasConflict implements VersionSpecification
+{
+    /** @var ContentService */
+    private $contentService;
+
+    /**
+     * @param ContentService $contentService
+     */
+    public function __construct(ContentService $contentService)
+    {
+        $this->contentService = $contentService;
+    }
+
+    /**
+     * Checks if $content has version conflict.
+     *
+     * @param VersionInfo $versionInfo
+     *
+     * @return bool
+     *
+     * @throws UnauthorizedException
+     * @throws BadStateException
+     */
+    public function isSatisfiedBy(VersionInfo $versionInfo): bool
+    {
+        if (!$versionInfo->isDraft()) {
+            throw new BadStateException('Version status', 'status is not draft');
```

Comment:
I don't think `Specification` should throw any exceptions. You are definitely checking too many things here. Please implement separate class for checking if the version is draft (so that we can reuse it in the future) and chain those specification objects together.